### PR TITLE
Restructure Project Configuration docs and add @import directive documentation

### PR DESCRIPTION
**Pull request by AnthonyMDev**
- Break the project configuration into multiple pages and expands the information on those pages. 
- Add "Client Directives" API Reference page
- Add "Sharing fragments across modules" section to operation models docs to explain usage of `@import` directive

### CODE REVIEW

File: docs/source/project-configuration/intro.mdx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+GraphQL schemas use a type system to define the shape of your available data. GraphQL operations (and your generated operation models) depend on a schema to know how to shape their operations to access a server's data.
+
+Apollo iOS generates a collection of types that provide the properties and metadata about each type in your GraphQL schema (i.e., objects, interfaces, unions, enums, input objects, etc.). Your generated [operation models](#operation-models) reference these shared schema types to provide type information without duplicating code.
+
+#### Test mocks
+
+Apollo iOS enables you to generate test mock objects that you can use in your test targets. These mocks enable you to create operation models for your tests.
+
+For more details, see [Test Mocks](./testing/test-mocks).
```
Comment (calvincestari) [inline docs/source/project-configuration/intro.mdx:40]:
```suggestion
For more details, see [Test Mocks](./../testing/test-mocks).
```
Same here.

File: docs/source/project-configuration/intro.mdx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+#### Operation models
+
+Apollo iOS generates _operation models_ from the files where you define your GraphQL operations (i.e., queries, mutations, subscriptions, and fragments).
+
+Operation models are classes that represent your GraphQL operations, which you can then use to make GraphQL requests (via `ApolloClient`).
+
+Each operation model includes a set of type-safe response model objects, representing that operation's result. When `ApolloClient` successfully executes a GraphQL request, it returns an instance of the corresponding response model for the executed operation.
+
+> To learn more about GraphQL operations, check out [Defining operations](./fetching/fetching-data#defining-operations).
```
Comment (calvincestari) [inline docs/source/project-configuration/intro.mdx:28]: This link doesn't work, I think it should be.
```suggestion
> To learn more about GraphQL operations, check out [Defining operations](./../fetching/fetching-data#defining-operations).
```

File: docs/source/project-configuration/modularization.mdx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+A single target project configuration means that all of your interaction with Apollo will be done from a single target in your project. This is usually the case if your project consists of a single monolithic application target.
+
+Alternatively you may use a single target configuration with a multi-module project when encapsulating your dependency on Apollo iOS behind a custom wrapper module. While it is common practice to encapsulate the setup of your `ApolloClient` and/or execution of operations, if you would like to use the generated operation response models in your other modules, you will need to use a multi-module project configuration.
+
+---
+
+For a single target project configuration, you can link your target to the `Apollo` library. You can also optionally link to [`ApolloSQLite`](#apollosqlite) and/or [`ApolloWebSocket`](#apollowebsocket) to enable their functionality.
+
+> For more information about the libraries that make up the Apollo iOS SDK, see the [Apollo iOS SDK](#apollo-ios-sdk-components).
```
Comment (calvincestari) [inline docs/source/project-configuration/modularization.mdx:24]: SDK components are now in a separate page, think it should be:
```suggestion
> For more information about the libraries that make up the Apollo iOS SDK, see the [Apollo iOS SDK](./apollo-ios-sdk-components).
```

File: docs/source/project-configuration/operation-models.mdx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+With relative paths, you can:
+- Co-locate models alongside the feature code using them
+- Include models in different modules across your project
+- Organize models based on feature area
+- Or use any other structure your project needs
+
+<Note>
+
+When including your operation models in a **multi-module project configuration**, ensure that any modules that include your operation models link to **both** your [schema types module](#schema-types-module) and the [`ApolloAPI` library](#apolloapi).
```
Comment (calvincestari) [inline docs/source/project-configuration/operation-models.mdx:36]:
```suggestion
When including your operation models in a **multi-module project configuration**, ensure that any modules that include your operation models link to **both** your [schema types module](./schema-types) and the [`ApolloAPI` library](./sdk-components/#apolloapi).
```

**Review by calvincestari — DISMISSED** [review]
Looks good, thank you! There are a couple links that need to be fixed.

File: docs/source/project-configuration/intro.mdx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+You can include your GraphQL models in your main application target, spread them across many modules, or expose them as an independent module that multiple applications can use. Depending on the needs and constraints of your project, you'll configure the code generation engine and link to the Apollo iOS libraries differently.
+
+There are three primary decisions you'll need to make about your project structure when using Apollo iOS:
+
+1. Will your project interact with Apollo in a single target or through multiple modules?
+2. Which schema `moduleType` should you use?
+3. Where should your generated operation models be located?
+
+Over the next few pages, we will cover the each of these questions and the possible choices for each.
```
Comment (BobaFetters) [inline docs/source/project-configuration/intro.mdx:54]:
```suggestion
Over the next few pages, we will cover each of these questions and the possible choices for each.
```

File: docs/source/project-configuration/operation-models.mdx [hunk 6]
```diff
@@ -0,0 +1,64 @@
+---
+title: Operation Models Configuration
+description: Configuring operation models for your projects needs
+---
+
+Operation models are generated for each of the GraphQL definitions defined in your project. These models are used to execute operations using the `ApolloClient` and include response models for accessing the data returned by these operation.
```
Comment (BobaFetters) [inline docs/source/project-configuration/operation-models.mdx:6]:
```suggestion
Operation models are generated for each of the GraphQL definitions defined in your project. These models are used to execute operations using the `ApolloClient` and include response models for accessing the data returned by these operations.
```

**Review by BobaFetters — APPROVED** [review]
Looks good to me other than the suggested changes, nice work!

**Pull request merged**